packages/@lwc/engine-server/src/apis/hot-swaps.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+/**
+ * These swapComponent is equivalent to the { swapComponent } functions exposed by @lwc/engine-dom. It doesn't do anything on
+ * the server side, however, you may import it.
+ *
+ * The whole point of defining this and exporting it is so that you can import it in isomorphic code without
+ * an error being thrown by the import itself.
+ * @throws Always throws, as it should not be used.
+ */
+export function swapComponent(): void {
+    if (process.env.NODE_ENV !== 'production') {
+        throw new Error(
+            'swapComponent is not supported in @lwc/engine-server, only @lwc/engine-dom.'
+        );
+    }
```

Comment:
I would avoid this `process.env.NODE_ENV` check – if the function is called, we should just throw an error in all environments. There is absolutely no reason someone should call this for the server. And for `engine-server` especially, there isn't really a strong difference between prod/development like there is for the browser.